bin/kmstool-enclave-cli/main.c

```diff
@@ -0,0 +1,218 @@
+    const struct aws_string *aws_secret_access_key;
+    const struct aws_string *aws_session_token;
+
+    const struct aws_string *ciphertext_b64;
+};
+
+static void s_usage(int exit_code) {
+    fprintf(stderr, "usage: enclave_server [options]\n");
+    fprintf(stderr, "\n Options: \n\n");
+    fprintf(stderr, "    --region REGION: AWS region to use for KMS\n");
+    fprintf(stderr, "    --proxy-port PORT: Connect to KMS proxy on PORT. Default: 8000\n");
+    fprintf(stderr, "    --aws-access-key-id ACCESS_KEY_ID: AWS access key ID\n");
+    fprintf(stderr, "    --aws-secret-access-key SECRET_ACCESS_KEY: AWS secret access key\n");
+    fprintf(stderr, "    --aws-session-token SESSION_TOKEN: Session token associated with the access key ID\n");
+    fprintf(stderr, "    --ciphertext CIPHERTEXT: base64-encoded ciphertext that need to decrypt\n");
+    fprintf(stderr, "    --help: Display this message and exit");
+    exit(exit_code);
+}
+
+static struct aws_cli_option s_long_options[] = {
+    {"region", AWS_CLI_OPTIONS_REQUIRED_ARGUMENT, NULL, 'r'},
+    {"proxy-port", AWS_CLI_OPTIONS_REQUIRED_ARGUMENT, NULL, 'x'},
+    {"help", AWS_CLI_OPTIONS_NO_ARGUMENT, NULL, 'h'},
+    {"aws-access-key-id", AWS_CLI_OPTIONS_REQUIRED_ARGUMENT, NULL, 'k'},
+    {"aws-secret-access-key", AWS_CLI_OPTIONS_REQUIRED_ARGUMENT, NULL, 's'},
+    {"aws-session-token", AWS_CLI_OPTIONS_REQUIRED_ARGUMENT, NULL, 't'},
+    {"ciphertext", AWS_CLI_OPTIONS_REQUIRED_ARGUMENT, NULL, 'c'},
+    {NULL, 0, NULL, 0},
+};
+
+static void s_parse_options(int argc, char **argv, struct app_ctx *ctx) {
+    ctx->proxy_port = PROXY_PORT;
+    ctx->region = NULL;
+
+    while (true) {
+        int option_index = 0;
+        int c = aws_cli_getopt_long(argc, argv, "r:x:k:s:t:c:h", s_long_options, &option_index);
+        if (c == -1) {
+            break;
+        }
+
+        switch (c) {
+            case 0:
+                break;
+            case 'r': {
+                ctx->region = aws_string_new_from_c_str(ctx->allocator, aws_cli_optarg);
+                break;
+            }
+            case 'x':
+                ctx->proxy_port = atoi(aws_cli_optarg);
+                break;
+            case 'k':
+                ctx->aws_access_key_id = aws_string_new_from_c_str(ctx->allocator, aws_cli_optarg);
+                break;
+            case 's':
+                ctx->aws_secret_access_key = aws_string_new_from_c_str(ctx->allocator, aws_cli_optarg);
+                break;
+            case 't':
+                ctx->aws_session_token = aws_string_new_from_c_str(ctx->allocator, aws_cli_optarg);
+                break;
+            case 'c':
+                ctx->ciphertext_b64 = aws_string_new_from_c_str(ctx->allocator, aws_cli_optarg);
+                break;
+            case 'h':
+                s_usage(0);
+                break;
+            default:
+                fprintf(stderr, "Unknown option\n");
+                s_usage(1);
+                break;
+        }
+    }
```

Comment:
Here you have to enforce that `ctx->aws_access_key_id` and `ctx->aws_secret_access_key` are set.

Reply:
Added codes to check if all required parameters exist